test/test_interpolations.jl

```diff
@@ -47,19 +47,29 @@
         # Note that not every element formulation exists for every order and dimension.
         applicable(Ferrite.getlowerorder, interpolation) && @test typeof(Ferrite.getlowerorder(interpolation)) <: Interpolation{ref_shape,func_order-1}
 
-    # Check partition of unity at random point.
-    n_basefuncs = getnbasefunctions(interpolation)
-    x = rand(Tensor{1, ref_dim})
-    f = (x) -> [shape_value(interpolation, Tensor{1, ref_dim}(x), i) for i in 1:n_basefuncs]
-    @test vec(ForwardDiff.jacobian(f, Array(x))') ≈
-        reinterpret(Float64, [shape_gradient(interpolation, x, i) for i in 1:n_basefuncs])
-    @test sum([shape_value(interpolation, x, i) for i in 1:n_basefuncs]) ≈ 1.0
-
-    # Check if the important functions are consistent
-    coords = Ferrite.reference_coordinates(interpolation)
-    @test length(coords) == n_basefuncs
-    @test shape_value(interpolation, x, n_basefuncs) == shape_value(interpolation, x, n_basefuncs)
-    @test_throws ArgumentError shape_value(interpolation, x, n_basefuncs+1)
+        n_basefuncs = getnbasefunctions(interpolation)
+        coords = Ferrite.reference_coordinates(interpolation)
+        f = (x) -> [shape_value(interpolation, Tensor{1, ref_dim}(x), i) for i in 1:n_basefuncs]
```

Comment:
Isn't this literally the same? I understand the recommendation above e.g. as not doing `ForwardDiff.jacobian(x->f(x), Array(x))')` instead of `ForwardDiff.jacobian(f, Array(x))')`, or what am I missing here?

Reply:
No, not literally, and it is in the test so not crucial, but this makes `f` not being a `const`, which is bad for type-stability etc. The recommendation I guess is just because (1) it means compiling another specialization if used in multiple places for the same `f` and (2) because it looks ugly :)

Reply:
Interesting. Can you add this to https://docs.julialang.org/en/v1/manual/style-guide/#Do-not-write-x-f(x) ? I think this is quite a crucial piece of information missing in the docs then.

Reply:
I'm not sure how crucial, because I think it only matters when you are working in the global scope (as we do here during testing), e.g. 
```julia
julia> using BenchmarkTools

julia> f(x) = sin(x)
f (generic function with 1 method)

julia> g = x -> sin(x)
#3 (generic function with 1 method)

julia> function foo(x)
           foofun(y) = sin(y)
           return foofun(x)
       end
foo (generic function with 1 method)

julia> function bar(x)
           barfun = y -> sin(y)
           return barfun(x)
       end
bar (generic function with 1 method)

julia> @btime f(1.0);
  0.800 ns (0 allocations: 0 bytes)

julia> @btime g(1.0); #Solved by @btime $g(1.0)
  19.659 ns (1 allocation: 16 bytes)

julia> @btime foo(1.0);
  0.800 ns (0 allocations: 0 bytes)

julia> @btime bar(1.0);
  0.800 ns (0 allocations: 0 bytes)
```